### PR TITLE
Legacy FSE: remove all Calypso references

### PR DESCRIPTION
**Pull request by mattwiebe**
#### Changes proposed in this Pull Request

* Legacy FSE is being cleansed from our system

#### Testing instructions

This touches a lot of things so hopefully all unit and e2e tests pass

Some manual testing would help!

Related to #56975

### CODE REVIEW

File: client/state/data-layer/wpcom/theme-filters/index.js [hunk 1]
```diff
@@ -34,16 +31,7 @@ const themeFiltersHandlers = dispatchRequest( {
...
-			const selectedSiteId = getSelectedSiteId( state );
-			const isFse = isSiteEligibleForLegacyFSE( state, selectedSiteId );
-			const isCoreFse = isSiteUsingCoreSiteEditor( state, selectedSiteId );
-
-			return themeFiltersHandlers( store, {
-				...action,
-				isFse,
-				isCoreFse,
-			} );
+			return themeFiltersHandlers( store, action );
```
Comment (Addison-Stavlo) [inline client/state/data-layer/wpcom/theme-filters/index.js:34]: should we continue to add the `isCoreFse` info here?

Comment (mattwiebe) [reply client/state/data-layer/wpcom/theme-filters/index.js:34]: Nope, it's no longer being used at all. It was, for a time, used inside the `storeFilters` `onSuccess` callback to filter out, um, filters, that we didn't want to show (aka back when we didn't show `block-templates` to Classic sites).

File: client/my-sites/domains/components/domain-warnings/index.jsx [hunk 2]
```diff
@@ -621,7 +621,7 @@ export class DomainWarnings extends PureComponent {
 			( { registrationDate } ) =>
 				registrationDate && moment( registrationDate ).add( 2, 'days' ).isAfter()
 		);
-		if ( this.props.isSiteEligibleForFSE && this.props.siteIsUnlaunched && isWithinTwoDays ) {
+		if ( this.props.siteIsUnlaunched && isWithinTwoDays ) {
```
Comment (Addison-Stavlo) [inline client/my-sites/domains/components/domain-warnings/index.jsx]: It seems previously this condition was only met for FSE eligible sites that were unlaunched, now with removing that first check this condition is met for all sites that are unlaunched?

Im not entirely sure what makes the most sense here: removing this entire conditional block (probably?), leaving it in with some legacy FSE active check, or other?

Comment (mattwiebe) [reply client/my-sites/domains/components/domain-warnings/index.jsx]: Thanks for checking! I will investigate, but my hunch is that complete removal is the way.

Comment (mattwiebe) [reply client/my-sites/domains/components/domain-warnings/index.jsx]: Chased this down to #37960 and you're right, I get to delete moar things ␡

**Review by Addison-Stavlo — APPROVED** [review]

**Pull request merged**